File: experiments/python_rewrite/seed_range/seed_range.py
from seed_map_layer import seed_map_layer



class SeedRange:

    

    def __init__(self, start: int, size: int):
        self.start = start
        self.end = start + size
        self.size = size
        pass

    def value_in_range(self, value):
        if value >= self.start and value < self.end:
            return True
        return False

    def find_min_in_range(self, seed_map_layers):
        
        min_value = 10**30
        for seed in range(self.start, self.end):
            value = seed
            for seed_map_layer in seed_map_layers:
                value = seed_map_layer.map_seed(value)
            if value < min_value:
                min_value = value

        return min_value

File: 2023/day05/c_sharp/app_code/SeedRange.cs
using System.Diagnostics;
using seed_map_layer;


namespace seed_range
{


    public class SeedRange
    {
        UInt64 start_;
        UInt64 size_;

        public UInt64 Start {
            get => start_;
        }

        public UInt64 End
        {
            get => start_ + size_;
        }

        public UInt64 Size
        {
            get => size_;
        }

        public SeedRange(UInt64 start, UInt64 size)
        {
            start_ = start;
            size_ = size;
        }
    }
}